cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -668,23 +668,41 @@ using SimpleArrayUint64 = SimpleArray<uint64_t>;
 using SimpleArrayFloat32 = SimpleArray<float>;
 using SimpleArrayFloat64 = SimpleArray<double>;
 
-enum class DataType
+class DataType
 {
-    Undefined,
-    Bool,
-    Int8,
-    Int16,
-    Int32,
-    Int64,
-    Uint8,
-    Uint16,
-    Uint32,
-    Uint64,
-    Float32,
-    Float64,
-}; /* end enum class DataType */
-
-DataType get_data_type_from_string(const std::string & data_type_string);
+public:
+    enum DataTypeEnum
+    {
+        Undefined,
+        Bool,
+        Int8,
+        Int16,
+        Int32,
+        Int64,
+        Uint8,
+        Uint16,
+        Uint32,
+        Uint64,
+        Float32,
+        Float64,
+    }; /* end enum Type */
```

Comment:
Please pay attention to end marking.  You are not consistent to the existing style nor your own code.